### ISSUE TITLE
wrong type inferred when (*args: T) -> Tuple[T, ...] is applied to Tuple[*Ts]

### ISSUE THREAD

**Issue by pinterior** [microsoft/pyright#4252]
**Describe the bug**

applying `(*args: T) -> Tuple[T, ...]` to `Tuple[*Ts]` results `Tuple[*Ts, ...]`, not the expected one `Tuple[Union[*Ts], ...]`

**To Reproduce**

```python
from typing import TypeVar, Tuple
from typing_extensions import reveal_type, TypeVarTuple

T = TypeVar("T")
Ts = TypeVarTuple("Ts")

def f(*args: T) -> Tuple[T, ...]:
    return args

reveal_type(f)

def g(x: "Tuple[*Ts]"):
    r = f(*x)
    reveal_type(r)
    return [i for i in r]

reveal_type(g)
```
```
n.py:10:13 - information: Type of "f" is "(*args: T@f) -> Tuple[T@f, ...]"
n.py:14:17 - information: Type of "r" is "Tuple[*Ts@g, ...]"
n.py:12:18 - warning: TypeVar "Ts" appears only once in generic function signature (reportInvalidTypeVarUse)
n.py:17:13 - information: Type of "g" is "(x: Tuple[*Ts@g]) -> list[object]"```
```

**Expected behavior**

- `r` in 'g' should have type `Tuple[Union[*Ts], ...]`
- return type of `g` should be `list[Union[*Ts]]`

**VS Code extension or command-line**

command-line tool, 1.1.282

**Comment by erictraut** [comment]
This will be addressed in the next release.

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.283, which I just published. It will also be included in a future version of pylance.
